src/Util/HTTPSignature.php

```diff
@@ -267,19 +268,18 @@ private static function decryptSigheader(array $headers, string $prvkey): string
 	/**
 	 * Post given data to a target for a user, returns the result class
 	 *
-	 * @param array   $data   Data that is about to be send
+	 * @param array   $data   Data that is about to be sent
 	 * @param string  $target The URL of the inbox
 	 * @param integer $uid    User id of the sender
 	 *
 	 * @return ICanHandleHttpResponses
-	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 * @throws HTTPException\NotFoundException
 	 */
 	public static function post(array $data, string $target, int $uid): ICanHandleHttpResponses
 	{
 		$owner = User::getOwnerDataById($uid);
-
 		if (!$owner) {
-			return null;
+			throw new HTTPException\NotFoundException('Owner data not found for ID ' . $uid);
```

Comment:
Where is this exception catched? Wouldn't such an exception stop worker executions which isn't the intended behaviour, would it?

Reply:
It would but then again we were implicitly throwing a `TypeError` exception before, now we just have more data about it.

Reply:
The function is only called at two places. We can easily catch it there. This error can only be thrown, when a user had recently been deleted and there had been a bunch of waiting worker processes for this user, which is some data constellation at single systems, but no indicator of a general problem. We can simply catch the error at the two places, I think.

Reply:
Turns out it wasn't that easy, before some callers of `HTTPSignature->post` were interpreting a failure as an incentive to retry sending the message. I quickly reached an unmanageable point with exception, so instead I worked my way back from the origin of the calls, retrieved the owner record there, and passed it down the call chain up to and including `HTTPSignature->post`.

Overall it reduces the number of calls to `User::getOwnerDataById` since we're passing fully-formed records around, and we don't have to care about whether the record exists or not deep down the call chain anymore.